tests/base/src/test/java/org/keycloak/test/admin/userprofile/UserProfileAdminTest.java

```diff
@@ -0,0 +1,142 @@
+package org.keycloak.test.admin.userprofile;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.keycloak.admin.client.resource.UserProfileResource;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserProfileAttributeGroupMetadata;
+import org.keycloak.representations.idm.UserProfileMetadata;
+import org.keycloak.representations.userprofile.config.UPAttribute;
+import org.keycloak.representations.userprofile.config.UPConfig;
+import org.keycloak.representations.userprofile.config.UPGroup;
+import org.keycloak.test.framework.annotations.InjectRealm;
+import org.keycloak.test.framework.annotations.KeycloakIntegrationTest;
+import org.keycloak.test.framework.injection.LifeCycle;
+import org.keycloak.test.framework.realm.ManagedRealm;
+import org.keycloak.test.util.JsonTestUtils;
+import org.keycloak.userprofile.config.UPConfigUtils;
+
+import java.util.List;
+import java.util.Map;
+
+@KeycloakIntegrationTest
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class UserProfileAdminTest {
+
+    @InjectRealm(lifecycle = LifeCycle.CLASS)
+    ManagedRealm realm;
```

Comment:
```suggestion
    private ManagedRealm realm;
```

Reply:
Better to leave without a modifier IMO; with `private` the IDE will complain that the field is never assigned.